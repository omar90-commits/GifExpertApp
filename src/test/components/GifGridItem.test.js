import React from 'react';
import { GifGridItem } from '../../components/GifGridItem';
import { shallow } from 'enzyme';

describe('Pruebas en el <GifGridItem />', () => {
	
	const title = 'Titulo de la App';
	const url = 'https://www.npmjs.com/package/enzyme-to-json';
	const wrapper = shallow( <GifGridItem title={title} url={url} /> );

	test('Debe de mostrar el componente correctamente', () => {
		
		expect( wrapper ).toMatchSnapshot();
	});

	test('Debe de tener un parrafo cone el titulo', () => {
		
		const p = wrapper.find('p');
		expect( p.text().trim() ).toBe( title );
	});

	test('Debe de tener la imagen igual al url y alt de los props', () => {
		
		const img = wrapper.find('img');

		expect( img.prop('src') ).toBe( url );
		expect( img.prop('alt') ).toBe( title );
	});

	test('Debe de tener la clase animate__fadeIn', () => {
		
		const div = wrapper.find('div');

		expect( div.hasClass('animate__fadeIn') ).toBe( true );
	});
});