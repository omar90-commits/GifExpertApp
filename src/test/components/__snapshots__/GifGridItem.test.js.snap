// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <GifGridItem /> Debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Titulo de la App"
    src="https://www.npmjs.com/package/enzyme-to-json"
  />
  <p>
     
    Titulo de la App
     
  </p>
</div>
`;
